File: all-submissions/0394-decode-string/solution.java
class Solution {
    public String decodeString(String s) {
        Stack<Integer> counts = new Stack();
        Stack<String> character = new Stack();
        int index = 0;
        String res = "";
        while(index< s.length()){
            if(Character.isDigit(s.charAt(index))){
                int count = 0;
                while(Character.isDigit(s.charAt(index))){
                    count = 10 * count +(s.charAt(index)-'0');
                    index++;
                }
                counts.push(count);
                
            } else if(s.charAt(index)=='[') {
                character.push(res);
                res = "";
                index++;
                
            } else if(s.charAt(index)==']') {
                StringBuilder temp = new StringBuilder();
                temp.append(character.pop());
                int count = counts.pop();
                while(count>0)
                {
                    temp.append(res);
                    count--;
                }
                res = temp.toString();
                index++;
                
            } else {
                res += s.charAt(index);
                index++;
            }
        }
        return res;
    }
}

File: all-submissions/0394-decode-string/solution.py
class Solution:
    def decodeString(self, s: str) -> str:
        stack = []
        ans = []
        for ch in s:
            if ch.isdigit():
                stack.append(ch)
            if ch.isalpha():
                stack.append(ch)
            if ch == "[":
                stack.append(ch)
            if ch == "]":
                curr_s = ""
                count = 0
                while stack and stack[-1] != "[":
                    curr_s = stack.pop() + curr_s
                stack.pop()
                base = 1
                while stack and stack[-1].isdigit():
                    count = count + int(stack.pop())*base
                    base *=10
                # ans += [curr_s]* count 
                stack.append(curr_s*count)
        if stack:
            curr_str = ""
            while stack:
                curr_str = stack.pop() + curr_str
            ans += [curr_str]
        return "".join(ans)


